app/controllers/course/learning_map_controller.rb

```diff
@@ -120,18 +131,27 @@ def map_condition_to_parent(condition)
     { id: id, is_satisfied: typed_condition.satisfied_by?(current_course_user) }
   end
 
-  def generate_nodes_from_conditionals(all_node_relations)
+  def generate_nodes_from_conditionals(all_node_relations) # rubocop:disable Metrics/AbcSize
     node_ids_to_children = all_node_relations[:node_ids_to_children]
     node_ids_to_parents = all_node_relations[:node_ids_to_parents]
+    node_ids_to_unlock_level = all_node_relations[:node_ids_to_unlock_level]
+    students = current_course.course_users.students
 
     @conditionals.map do |conditional|
       id = get_node_id(conditional)
+      num_students_unlocked = 0
+      students.each do |student|
+        num_students_unlocked += 1 if conditional.conditions_satisfied_by?(student)
+      end
+      total_num_students = current_course.course_users.students.count
```

Comment:
This can be shifted out of the loop and changed to

```ruby
total_num_students = students.count
```

Reply:
Done.